### PR TITLE
[SPARK-25705][BUILD][STREAMING][test-maven] Remove Kafka 0.8 integration

### PR DESCRIPTION
**Pull request by srowen**
## What changes were proposed in this pull request?

Remove Kafka 0.8 integration

## How was this patch tested?

Existing tests, build scripts

### CODE REVIEW

File: python/pyspark/streaming/tests.py [hunk 1]
```diff
@@ -1047,259 +1046,6 @@ def check_output(n):
         self.ssc.stop(True, True)
 
 
-class KafkaStreamTests(PySparkStreamingTestCase):
```
Comment (srowen) [inline python/pyspark/streaming/tests.py]: Am I correct that all of this Pyspark Kafka integration is 0.8, not 0.10? that structured streaming is the only option now for Pyspark + Kafka?

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
retest this please

File: python/pyspark/streaming/tests.py (hunk 1 above)
Comment (srowen) [reply python/pyspark/streaming/tests.py]: OK, you or @holdenk or @koeninger might want to skim this change to make sure I didn't delete Pyspark + Structured Streaming + Kafka support inadvertentently. I don't think so, but it's not my area so much.

File: docs/streaming-kafka-0-10-integration.md [hunk 2]
```diff
@@ -3,7 +3,11 @@ layout: global
 title: Spark Streaming + Kafka Integration Guide (Kafka broker version 0.10.0 or higher)
 ---
 
-The Spark Streaming integration for Kafka 0.10 is similar in design to the 0.8 [Direct Stream approach](streaming-kafka-0-8-integration.html#approach-2-direct-approach-no-receivers).  It provides simple parallelism,  1:1 correspondence between Kafka partitions and Spark partitions, and access to offsets and metadata. However, because the newer integration uses the [new Kafka consumer API](http://kafka.apache.org/documentation.html#newconsumerapi) instead of the simple API, there are notable differences in usage. This version of the integration is marked as experimental, so the API is potentially subject to change.
+The Spark Streaming integration for Kafka 0.10 provides simple parallelism, 1:1 correspondence between Kafka 
+partitions and Spark partitions, and access to offsets and metadata. However, because the newer integration uses 
+the [new Kafka consumer API](https://kafka.apache.org/documentation.html#newconsumerapi) instead of the simple API, 
+there are notable differences in usage. This version of the integration is marked as experimental, so the API is
```
Comment (koeninger) [inline docs/streaming-kafka-0-10-integration.md]: Do we want to leave the new integration marked as experimental if it is now the only available one?

Comment (srowen) [reply docs/streaming-kafka-0-10-integration.md]: Yeah, good general point. Is the kafka 0.10 integration at all experimental anymore? Is anything that survives from 2.x to 3.x? I'd say "no" in almost all cases. What are your personal views on that?

**Comment by koeninger** [comment]
I guess the only argument to the contrary would be if some of the known
issues end up being better solved with minor API changes, leaving it marked
as experimental would technically be better notice.

I personally think it's clearer to remove the experimental.

File: python/pyspark/streaming/tests.py (hunk 1 above)
Comment (HyukjinKwon) [reply python/pyspark/streaming/tests.py]: I skimmed and seems fine. Will try to take a look few times more while it's open. (don't block by me)

**Comment by srowen** [comment]
So far looking good to those who have looked, and it passed Maven and SBT tests. I think this will help reduce complexity a bit (and test time in some cases), so will go for it tomorrow.

**Pull request closed without merging**